### ISSUE TITLE
Add log and document prover behavior when claim has `\\bottom` RHS

### ISSUE THREAD

**Issue by rod-lin** [kframework/kore#2703]
Hello! This is probably an edge case but I think kprove is supposed to validate the `A-SPEC` spec below:
```
module A

syntax Foo ::= "a"

rule a => a

endmodule
```

```
module A-SPEC

imports A

claim a => . ensures false

endmodule
```

But instead it gets stuck at
```
kore-exec: [165966] Warning (WarnStuckClaimState):
    (InfoReachability) while checking the implication:
    The proof has reached the final configuration, but the claimed implication is not valid. Location: /home/rodlin/Desktop/illinois/research/matching-logic-prover/matching-logic-proof-checker/examples/imp/.kprove-2021-06-26-19-29-45-382-5bb92c7f-1d1b-4273-a67b-df0c336a1d60/././a-spec.k:5:7-5:27
<k>
  a ~> _DotVar1 ~> .
</k>
```

I think perhaps it applied the semantics rule first before applying the claim inductively?
The java backend is able to prove this.

**Comment by ana-pantilie** [comment]
I discussed this issue with @ttuegel, the conclusion is this: we don't support attempting to prove claims which have `\\bottom` RHSs. We currently stop the proof if we detect that the RHS of the claim is `\\bottom`. If we didn't do that, in the general case we would end up looping forever.

Thanks for opening this issue, though, because we realized that we should add a section in [K's pending documentation](https://github.com/kframework/k/blob/master/pending-documentation.md) about this.
